lib/xboxrt/libc_extensions/direct.c

```diff
@@ -46,26 +46,28 @@ static int convert_error(DWORD winerror)
  */
 
 int _chdir(char* path) {
-    return 0;
+    assert(0);
+    return -1; // Unreachable
 }
 
 int _chdrive(int drive) {
-    return 0;
+    assert(0);
+    return -1; // Unreachable
 }
 
 char* _getcwd(char* buffer, size_t length) {
-    errno = -EINVAL;
-    return NULL;
+    assert(0);
+    return NULL; // Unreachable
```

Comment:
I believe we can hard-code this for now, so at least apps which use this to find files will work (sometimes some SDK or API will require absolute paths, and then the application might query the base path through this).

I can see how that would be a lot of additional implementation effort, so it can be done in a follow-up PR.

---

If this follows the same convention, we might be able to use: https://github.com/XboxDev/nxdk/blob/500354ef264f35a46a31643876b06ed614b0aed8/lib/nxdk/path.c#L9-L14

Otherwise, we might be able to hardcode it to `D:\\` which is either (I don't remember) a kernel requirement or at least a dashboard- or self-imposed XBE controlled symlink?

Eitherway, we'd have to follow conventions of the MS implementation (regarding forward vs backward slash, device names vs driver letters etc.).

Similar reasoning for `_getdrive`, although I speculate that this is required much less.

Reply:
D drive letter mount is handled by XDK on boot when D drive letter isn't set by the kernel. Otherwise, it is handled by the kernel on reboot.

Though, I highly recommended to check `LaunchDataPage->Header.szLaunchPath` first for semicolon character which is mount as working directory by the kernel itself. Then fallback to XeImageFileName with null terminate append at end of string.

fyi: I provided this type of information in XboxDev discord server based on what I had researched years ago.

Reply:
I'd prefer `XeImageFileName` (which the kernel uses to talk to the app / after launching) over `LaunchDataPage` (which is used by the app to talk to the kernel / before launching)

Reply:
Yes, LaunchDataPage is used to talk to the kernel but that's only half correct. It is also used by the app as well hence the LaunchData member variable from LAUNCH_DATA_PAGE structure is used to tell the app, either from same xbe or another xbe, what to do next after reboot occur. The kernel doesn't modify LaunchDataPage created by the app. The kernel read from `LaunchDataPage->Header` structure to do what the app want the kernel to do on reboot.

Here's a log of how it works on first boot and reboot:
```
---- first boot ----
LaunchDataPage is NULL.
XeImageFileName: \Device\Harddisk0\Partition1\Games\reboot\default.xbe
\??\D: == \Device\Harddisk0\Partition1\Games\reboot
Relaunching... LaunchDataPage->Header.szLaunchPath: \Device\Harddisk0\Partition1\Games;reboot\default.xbe
============
------ reboot ------
LaunchDataPage->Header.szLaunchPath: \Device\Harddisk0\Partition1\Games;reboot\default.xbe
XeImageFileName: \Device\Harddisk0\Partition1\Games\reboot\default.xbe
\??\D: == \Device\Harddisk0\Partition1\Games
============
```

As you can see XeImageFileName doesn't include semicolon which break working directory mount from D: symlink pointing to a parent directory than the same directory according to reboot process.
This is the same/similar concept on Windows' command prompt or powershell usage. When you change directory from the terminal itself, i.e. `C:\logs>` . Then execute `D:\build\UnitTest.exe` which will create log file into `C:\logs` instead of `D:\build` directory.

In summary, this is how demo discs and even xbox live works when they focus on D:\ symlink as a working directory. Otherwise attempting to access the contents wouldn't work engineered by original xbox team. I hope this is more clarified for you and everyone else reading this.